src/server.c

```diff
@@ -2063,14 +2063,16 @@ int serverCron(struct aeEventLoop *eventLoop, long long id, void *clientData) {
 
     /* AOF postponed flush: Try at every cron cycle if the slow fsync
      * completed. */
-    if (server.aof_flush_postponed_start) flushAppendOnlyFile(0);
+    if (server.aof_state == AOF_ON && server.aof_flush_postponed_start)
+        flushAppendOnlyFile(0);
 
     /* AOF write errors: in this case we have a buffer to flush as well and
      * clear the AOF error in case of success to make the DB writable again,
      * however to try every second is enough in case of 'hz' is set to
      * a higher frequency. */
     run_with_period(1000) {
-        if (server.aof_last_write_status == C_ERR)
+        if (server.aof_state == AOF_ON &&
+            server.aof_last_write_status == C_ERR)
```

Comment:
styling: when an `if` has two lines we need to open curly brace on a line of it's own, in this case i think one line is ok.
```suggestion
        if (server.aof_state == AOF_ON && server.aof_last_write_status == C_ERR)
```